src/main/scala/io/qbeast/spark/index/query/QueryFiltersUtils.scala

```diff
@@ -91,15 +96,25 @@ private[query] trait QueryFiltersUtils {
   }
 
   /**
-   * Convert an Spark String type to a Scala core type
-   * @param value the value to convert
+   * Convert an Literal value from Spark to a Qbeast/Scala core type
+   * @param l the Literal to convert
    * @return
    */
 
-  def sparkTypeToCoreType(value: Any): Any = {
-    value match {
-      case s: UTF8String => s.toString
-      case _ => value
+  def sparkTypeToCoreType(l: Literal): Any = {
+
+    (l.value, l.dataType) match {
+      case (int: Integer, _: DateType) =>
+        // convert DateType to Milliseconds
+        lazy val zoneId = getZoneId(SQLConf.get.sessionLocalTimeZone)
```

Comment:
I wonder that the data is always stored as the Spark session local timezone, right?

Reply:
Do you mean that the data is persisted on disk using the timezone of Spark?

Reply:
According to this post: https://www.databricks.com/blog/2020/07/22/a-comprehensive-look-at-dates-and-timestamps-in-apache-spark-3-0.html If no Timezone is specified, **Spark takes a time zone from the SQL configuration `spark.sql.session.timeZone` and applies it to function invocations**. 

So yes, data would be persisted on the Disk using the `spark.sql.session.timeZone`.

Reply:
My bad, after reading the article I think the code is correct.